common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java

```diff
@@ -48,6 +49,15 @@ public long size() {
   }
 
   /**
+   * Reset the size of the memory block.
+   */
+  public void resetSize(long len) {
+    assert (ByteArrayMethods.roundNumberOfBytesToNearestWord(length) ==
```

Comment:
We'd better use `require` instead of `assert`.

Reply:
Also leave some message if the check fails.

Reply:
It seems that `require`  can not be used in the Java code @jiangxb1987